foyer/forcefield.py

```diff
@@ -118,6 +118,53 @@ def _topology_from_parmed(structure, non_element_types):
     positions = structure.positions
     return topology, positions
 
+
+def _topology_from_residue(res):
+    """Converts a openmm.app.Topology.Residue to openmm.app.Topology"""
+    topology = app.Topology()
+    chain = topology.addChain()
+    new_res = topology.addResidue(res.name, chain)
+
+    atoms = dict()  # omm.Atom in res : omm.Atom in topology
+
+    for res_atom in res.atoms():
+        topology_atom = topology.addAtom(name=res_atom.name,
+                         #element=elem.Element.getBySymbol(res_atom.name),
+                         element=res_atom.element,
```

Comment:
Inside of ```apply```, ```_topology_from_residue``` acts on a topology that was created with ```generate_topology``` so this case should already be handled.